typescript/common-resolvers/src/utils/validate-upload-mime-types.ts

```diff
@@ -0,0 +1,87 @@
+import mime from "mime-types";
+import JSZip from "jszip";
+
+export enum ValidMimeTypeCategory {
+  image = "image",
+  archive = "archive",
+  json = "json",
+}
+
+export const validMimeTypes: Record<ValidMimeTypeCategory, string[]> = {
+  image: ["image/jpeg", "image/png", "image/bmp"],
+  archive: ["application/zip"],
+  json: ["application/json"],
+};
+
+export const validMimeTypesFlat: string[] = Object.values(
+  validMimeTypes
+).reduce<string[]>(
+  (mimeTypesFlat, mimeTypeList) =>
+    mimeTypeList.reduce<string[]>((mimeTypesFlatSub, mimeTypeString) => {
+      mimeTypesFlatSub.push(mimeTypeString);
+      return mimeTypesFlatSub;
+    }, mimeTypesFlat),
+  [] as string[]
+);
+
+export function getValidMimeTypeCategoriesList(
+  categories: ValidMimeTypeCategory[] | undefined = undefined
+): string[] {
+  return categories === undefined
+    ? validMimeTypesFlat
+    : Object.values(categories).reduce<string[]>(
+        (mimeTypesFlat, mimeTypeCategory) =>
+          validMimeTypes[mimeTypeCategory].reduce<string[]>(
+            (mimeTypesFlatSub, mimeTypeString) => {
+              mimeTypesFlatSub.push(mimeTypeString);
+              return mimeTypesFlatSub;
+            },
+            mimeTypesFlat
+          ),
+        [] as string[]
+      );
+}
+
+export function getValidMimeTypeCategoriesListString(
+  categories: ValidMimeTypeCategory[] | undefined = undefined
+): string {
+  return getValidMimeTypeCategoriesList(categories).join(", ");
+}
```

Comment:
I don't know if this one is really needed? I guess it's used only for the line bellow

Reply:
Yeah I sometimes have the bad habit of doing util functions "just in case" which will probably end up unused and "polluting" the code